src/SWP/Bundle/CoreBundle/EventListener/DuplicatedSlugListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Core Bundle.
+ *
+ * Copyright 2018 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2018 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\CoreBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Bundle\CoreBundle\Model\ArticleInterface;
+use SWP\Bundle\CoreBundle\Repository\ArticleRepositoryInterface;
+
+class DuplicatedSlugListener
+{
+    /**
+     * @var ArticleRepositoryInterface
+     */
+    protected $articleRepository;
+
+    /**
+     * @param ArticleRepositoryInterface $articleRepository
+     */
+    public function __construct(ArticleRepositoryInterface $articleRepository)
+    {
+        $this->articleRepository = $articleRepository;
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function onArticleCreate(ArticleEvent $event)
+    {
+        /** @var ArticleInterface $article */
+        $article = $event->getArticle();
+
+        $existingArticle = $this->articleRepository->findOneBy(['slug' => $article->getSlug()]);
+        if (null === $existingArticle) {
+            return;
+        }
+
+        $hash = substr(md5($article->getPackage()->getGuid()), 0, 8);
```

Comment:
you can use `RandomStringGenerator` for that suffix generation

Reply:
We don't want it to be random. This way we can predict that hash based on package guid (in tests, or in future cases).